src/models/Scholarships.ts

```diff
@@ -100,6 +98,13 @@ class Scholarships extends FirestoreCollection<ScholarshipData> {
 
     return FirestoreCollection.list(query, postProcessFilter);
   }
+
+  save(data: ScholarshipData): Promise<Model<ScholarshipData>> {
+    const Data = { ...data, lastModified: new Date() };
+    if (!data.dateAdded) Data.dateAdded = new Date();
+
+    return super.new(Data).save();
```

Comment:
I think this doesn't work for existing scholarships (you might try adding a test case that proves/disproves my theory). If you call `Scholarships.save(data)` on an existing scholarship `.new()` creates a new document with a new ID (hence the function name) so the scholarship wouldn't be saved under the same ID. If you call `scholarship.save()` on an existing scholarship then none of the logic here will run.

Now how to make this work, hmmm.  There's a couple options:
- Create a class that extends `FirestoreModel<Scholarship>` and overrides the `.save()` method with this business logic.
- Remove the `.save()` method from `FirestoreModel<Scholarship>` and accept a `FirestoreModel<Scholarship>` instead.

I'm not sure which is easier but in any case I recommend adding a test case to make sure you're implementing it correctly!